internal/prometheus.go

```diff
@@ -116,7 +116,8 @@ func (p Prometheus) convertTags(ss promCommon.Metric, c QueryConf) map[string]st
 func (p Prometheus) normalize(s string) string {
 	b := []byte(s)
 	for i, c := range b {
-		if !((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9') || (c >= 'A' && c <= 'Z')) {
+		if !((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9') || (c >= 'A' && c <= 'Z') ||
+			c == '-' || c == '_' || c == '.' || c == '/') {
```

Comment:
That's why I added checking code of `_`. The original code does it.

Reply:
Damn, I did not write what I was thinking about... sorry.

Do you agree that this code will also do the job:
```go
if !((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9') || (c >= 'A' && c <= 'Z')) || c == '-' || c == '.' || c == '/') {
```

The only difference is that:
- your code will check if the characted is `_`, and if it is, it will keep the character
- the code above will consider `_` as a blacklisted character... and replace it with `_`

I don't know what's the most efficient (for the specific `_` character):
- check and do nothing
- or replace without check

Reply:
I think writing to memory consumes more CPU cycles than just reading from memory. Plus, the code always reads `c` to check what value `c` is, but writing to the array is definitely additional operation. My conclusion is that removing the additional operation looks the most efficient.

Reply:
Let's forget about it, let's keep it this way.